### DISCUSSION TITLE
Self-hosting on Azure

### DISCUSSION THREAD

**Discussion by mellson** [supabase/supabase#5112] [Questions]
Is it possible to self host Supabase on Azure? I've stumbled upon a "small" roadblock in the form of `pgjwt` not being available in Azure Database for PostgreSQL. Has anyone tried to self-host on Azure and found a way around this problem?

**Comment by jordevorstenbosch** [comment]
I've been trying as well, and the only way to get it to work is to basically host the DB yourself (even if on a separate machine).

Documentation is misleading as it initially says:

> Using an external database
We strongly [recommend](https://supabase.com/docs/guides/hosting/overview#managing-your-database) that you decouple your database from docker-compose before deploying. The middleware will run with any PostgreSQL database that has logical replication enabled. The following environment variables should be updated in the .env file to point to your external database:

As well as:
>It is recommended that you decouple your database from the middleware so that you can upgrade the middleware without any downtime. The "middleware" is everything except Postgres, and it should work with any Postgres provider (such as AWS RDS), or your own Postgres cluster.

But in reality it indeed requires pgjwt as well as superuser and bypassrls which no major cloud provider seems to support. 

Even the mentioned AWS RDS does not support pgjwt if I'm to believe their documentation:
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.Extensions.Trusted=

I've only used DO and Azure personally, and I can't get supabase self-hosted to run with their managed DBs. So I guess we're stuck with IaaS for those scenarios.

**Comment by ElToro13** [comment]
Same issue here. pyjwt doesnt work even with AWS RDS. Their Documentation says AWS RDS is fully supported. But when you run their extensions creation commands, all works expect 
```
create extension if not exists pgjwt with schema extensions;
```

**Comment by steve-chavez** [comment]
Note that `pgjwt` is pure SQL, you can create it without doing `create extension`. Some instructions [here](https://postgrest.org/en/stable/auth.html#jwt-from-sql).

**Reply by jordevorstenbosch** [reply]
Even when you add pgjwt manually it still won’t run without all the extra permissions it requires.

**Comment by ElToro13** [comment]
Also like to add that Self hosting is not working for me with AWS RDS
Can't connect to supabase-auth
```cmd
supabase-auth | time="2022-07-15T14:09:51Z" level=fatal msg="running db migrations: Migrator: problem creating schema migrations: CREATE TABLE \"schema_migrations\" (\n\"version\" VARCHAR (14) NOT NULL\n);\nCREATE UNIQUE INDEX \"schema_migrations_version_idx\" ON \"schema_migrations\" (version);: ERROR: no schema has been selected to create in (SQLSTATE 3F000)"
supabase-auth | [POP] 2022/07/15 14:09:51 info - 2.6074 seconds
supabase-auth exited with code 1
```

**Reply by PrasannaSM** [reply]
@ElToro13 were you able to fix the issue of self-hosted supabase with RDS backend?

**Reply by ElToro13** [reply]
No.. had to rewrite my entire backend code

**Reply by PrasannaSM** [reply]
backend as in supabase auth image?

**Reply by ElToro13** [reply]
no. Removed supabase altogether and use Postgres directly

**Reply by PrasannaSM** [reply]
@ElToro13 Thats good. So, let me tell you my use case. The thing to vendor lock I am looking for an alternate to firebase realtime database [for live sync]. I stumbled upon supabase as an interim solution. I wanted to try out self-hosted supabase with AWS RDS. Now, facing with this error. Any suggestions

**Reply by ElToro13** [reply]
this was my first project with SQL. I usually worked with NoSQL (MongoDB). So really didnt know much about how to fix this error. I wrote a Stackover flow question regarding the same. Someone was kind enough to write a detailed answer. You can check it out [here](https://stackoverflow.com/questions/72999025/supabase-self-hosting-issue). Hope that helps!

**Comment by AnkityadavIITR** [comment]
can anyone help to deploy supabase on azure container apps

**Reply by zaynekomichi** [reply]
any update on this

**Reply by dheeraj-vp** [reply]
were you able to do it?

**Comment by mly-johndpope** [comment]
im making some progress - auth is working - i will share some more scripts to get it up once i fix elixir - 
https://gist.github.com/johndpope/62d3229c93bb0bb11d35b538251e8c45

**Comment by TalkSimple** [comment]
Hi I have the same question. What were peoples experiences? Did you get it to work or try something else?

**Comment by mly-johndpope** [comment]
i wasted a lot of time on this - we have auth working . realtime gave up on.

the supabase elixir code is incompatible with rds - 

i troubleshooot many errors on bootup - deep diving into containers auth / realtime

this script gets the base up and running. 
for analytics i just the docker db postgres. no backups / performance needed. 

https://github.com/johndpope/rds-supabase/blob/main/docker/bootstrap_db.sh

https://github.com/johndpope/realtime
i use the root / master account named as supabase_admin  - this helps. 
the other gotcha is the search path. 
ALTER ROLE supabase_admin SET search_path TO auth, public;

**Reply by zaynekomichi** [reply]
Thanks for the update

**Comment by Rubenburdin** [comment]
The short answer from what I've seen (and from mly-johndpope's experience above) is that self-hosting Supabase with a managed Postgres like Azure Database or AWS RDS remains painful. The root cause hasn't changed much: managed Postgres services don't give you the superuser or bypassrls permissions Supabase components expect, and pgjwt still isn't whitelisted on most providers.
If you really need Azure, the workaround is running Postgres yourself on a VM or container rather than using their managed offering. That gives you full control over extensions and roles. The bootstrap scripts mly-johndpope shared are a decent starting point for RDS, though realtime remains tricky because the Elixir code has assumptions about role permissions that managed DBs just don't satisfy.
One thing worth considering: if your goal is getting production data into Supabase from other systems (Salesforce, HubSpot, etc.), you might avoid some of this pain by using Supabase Cloud for the hosted bits and handling the external sync separately. At Stacksync we've seen teams go that route when self-hosting becomes too much overhead — though it depends on your compliance or data residency requirements.
What's driving the Azure requirement on your end? That might help narrow down whether the VM-based approach is realistic for your setup.
